lms/djangoapps/grades/models.py

```diff
@@ -302,6 +302,13 @@ def _is_unattempted_with_score(self):
         """
         return self.first_attempted is None and any(field != 0.0 for field in (self.earned_all, self.earned_graded))
 
+    def full_clean(self, exclude=None, validate_unique=True):
+        # efischer debug, nomerge
+        from nose.tools import set_trace; set_trace()
+        self.clean_fields(exclude='visible_blocks')
+        self.clean()
+        # validate_unique intentionally skipped here
```

Comment:
And `validate_unique` is the other, asking "Is there a record with the same `course_id`, `usage_id`, and `user_id`, but a different `id`?"
```
> /edx/app/edxapp/edx-platform/lms/djangoapps/grades/models.py(310)full_clean()
-> self.validate_unique()
(Pdb) n
DEBUG:django.db.backends:(0.001) SELECT (1) AS `a` FROM `grades_persistentsubsectiongrade` WHERE (`grades_persistentsubsectiongrade`.`course_id` = 'course-v1:edX+DemoX+Demo_Course' AND `grades_persistentsubsectiongrade`.`usage_key` = 'block-v1:edX+DemoX+Demo_Course+type@sequential+block@basic_questions' AND `grades_persistentsubsectiongrade`.`user_id` = 3 AND NOT (`grades_persistentsubsectiongrade`.`id` = 13)) LIMIT 1; args=(u'course-v1:edX+DemoX+Demo_Course', u'block-v1:edX+DemoX+Demo_Course+type@sequential+block@basic_questions', 3, 13)
```